winml/adapter/WinMLAdapter.cpp

```diff
@@ -0,0 +1,746 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "pch.h"
+#include "WinMLAdapter.h"
+#include "CustomRegistryHelper.h"
+#include "PheonixSingleton.h"
+#include "LotusEnvironment.h"
+#include "AbiCustomRegistryImpl.h"
+
+#ifdef USE_DML
+#include "core/providers/dml/DmlExecutionProvider/inc/DmlExecutionProvider.h"
+#include "core/providers/dml/GraphTransformers/GraphTransformerHelpers.h"
+#include "core/providers/dml/OperatorAuthorHelper/SchemaInferenceOverrider.h"
+#include "DmlOrtSessionBuilder.h"
+#endif USE_DML
+
+#include "LearningModelDevice.h"
+#include "TensorFeatureDescriptor.h"
+#include "ImageFeatureDescriptor.h"
+#include "api.image/inc/D3DDeviceCache.h"
+#include "Common/inc/WinMLTelemetryHelper.h"
+
+#include "CpuOrtSessionBuilder.h"
+
+#include <io.h>
+#include <fcntl.h>
+
+#include "ZeroCopyInputStreamWrapper.h"
+#include "google/protobuf/io/zero_copy_stream_impl.h"
+
+#include "FeatureDescriptorFactory.h"
+#include "core\framework\utils.h"
+#include "core\framework\session_state.h"
+
+using namespace winrt::Windows::AI::MachineLearning;
+
+namespace Windows::AI::MachineLearning::Adapter {
+
+// Define winml trace logging provider with WinML GUID
+TRACELOGGING_DEFINE_PROVIDER(
+    winml_trace_logging_provider,
+    WINML_PROVIDER_DESC,
+    WINML_PROVIDER_GUID);
+
+// ORT intentionally requires callers derive from their session class to access
+// the protected methods used below.
+class InferenceSessionProtectedLoadAccessor : public onnxruntime::InferenceSession {
+ public:
+  onnxruntime::common::Status
+  Load(std::unique_ptr<ONNX_NAMESPACE::ModelProto> p_model_proto) {
+    return onnxruntime::InferenceSession::Load(std::move(p_model_proto));
+  }
+  const onnxruntime::SessionState& GetSessionState() {
+    return session_state_;
+  }
+};
+
+class ModelProto : public Microsoft::WRL::RuntimeClass<
```

Comment:
do we want to split classes out into their own files? this adapter file is getting pretty large.

Reply:
yup !  add it to the list.  i hesistated because i wans't sure yet what the final refactor was going to look like .  once we are pretty sure (80%) that we are done with the layering (adapter vs. c_abi) then we should totes clean up our remaining adapter parts.

---
In reply to: [354039549](https://github.com/microsoft/onnxruntime/pull/2534#discussion_r354039549) [](ancestors = 354039549)

Reply:
i added it to your whiteboard

---
In reply to: [354617810](https://github.com/microsoft/onnxruntime/pull/2534#discussion_r354617810) [](ancestors = 354617810,354039549)